### ISSUE TITLE
Basic Hello World script from tutorial not accepting correctly formatted JSON

### ISSUE THREAD

**Issue by carinahealth-alden** [supabase/supabase#20887]
I'm just getting my supabase edge environments setup, and admittedly a bit of a code newbie.

I have gotten to the place where I'm successfully able to function serve and listen to the functions processed. I can update the provided hello world script to execute it all correctly beginning to end (so not an auth issue). Basically this works:

 `Deno.serve(async (req) => {
  const name = 'text'
  const data = {
    message: `Hello ${name}!`,
  }
return new Response(
    JSON.stringify(data),
    { headers: { "Content-Type": "application/json" } },
  )
})`

when I send the request with 

`curl -v --location --request POST 'myUrl' --header 'Authorization: Bearer myToken' --header 'Content-Type: application/json' --data '{ "name":"Functions" }'`

properly returns {"message":"Hello text!"}

But when I update back to the original/provided function from https://supabase.com/docs/guides/functions/quickstart, which is:

`Deno.serve(async (req) => {
  const { name } = await req.json()
  const data = {
    message: `Hello ${name}!`,
  }
return new Response(JSON.stringify(data), { headers: { 'Content-Type': 'application/json' } })
})
`

and then send the same request above I get a 500 ISE. Specifically, a `[Error] SyntaxError: Expected property name or '}' in JSON at position 2 (line 1 column 3)`

If I send the request with `--data '{}'` instead, it executes correctly with `{"message":"Hello undefined!"}` as the response. 

**So basically I dont understand why it's not accepting a (what seems to me to be) correctly formatted JSON**

**Comment by carinahealth-alden** [comment]
In short: I published the out-of-the-box QuickStart. When I hit the function with -- data ‘{}’ I get a successful 200 execution with {“message”:”Hello undefined!”} (which seems fine / as expected), but when I hit the function with --data ‘{“name”:”Functions”}’ (just what the QuickStart guide has) I get a 500 ISE for “Syntax Error: expected property name or ‘}’ in JSON at position 1”
Is there something I need to install so it can correctly process jsons? … Or why would a correctly formatted json not process correctly?

**Comment by carinahealth-alden** [comment]
ok this was just a dumb escape issue - need to format curl jsons with \ escapes.

--data '{\"name\":\"Functions\"}'
